packages/manager/apps/pci-quota/src/api/hooks/useRegions.ts

```diff
@@ -0,0 +1,77 @@
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { useTranslatedMicroRegions } from '@ovh-ux/manager-react-components';
+import { getProjectRegions } from '@ovh-ux/manager-pci-common';
+import { getAvailableRegions } from '@/api/data/region';
+
+type TLocation = {
+  continent: string;
+  regions: Set<string>;
+};
+
+type TPlainLocation = {
+  name: string;
+  continent: string;
+  regions: string[];
+};
+
+export const useGetRegions = (projectId: string) =>
+  useQuery({
+    queryKey: ['project', projectId, 'regions'],
+    queryFn: () => getProjectRegions(projectId),
+  });
+
+export const useGetAvailableRegions = (projectId: string) =>
+  useQuery({
+    queryKey: ['project', projectId, 'availableRegions'],
+    queryFn: () => getAvailableRegions(projectId),
+  });
+
+export const useLocations = (projectId: string, onlyAvailable = false) => {
+  const query = onlyAvailable
+    ? useGetAvailableRegions(projectId)
+    : useGetRegions(projectId);
```

Comment:
Hooks shouldn't be call depending on a condition, this should be split into 2 different calls or call one useQuery with parameters